### ISSUE TITLE
Android app: Two finger and one finger scroll should be swappable

### ISSUE THREAD

**Issue by freebrowser1** [iiordanov/remote-desktop-clients#405]
The bVNC app works awesome, but the much more often scroll screen function is bound to two-finger scroll and the less often screen pan function to one finger. But one finger operation is much more convenient.

Can there be made an option to bound one finger scroll to the scroll function and two finger only for screen pan function ?

**Comment by iiordanov** [comment]
There is another input mode where pan and drag and drop are swapped
already. I felt like that's more intuitive. However your suggestion does
have erit for certain people.

Have you noticed that in fit-to-screen scaling the app scrolls with one
finger already?

Iordan

**Comment by freebrowser1** [comment]
> Have you noticed that in fit-to-screen scaling the app scrolls with one
> finger already?
Yes, I tried this and it works, but then it is not possible to move the mouse arrow to the edge of the screen which is required for many actions (e.g. menu bar, exiting full screen in some environments). In that case, a two finger scroll should actually pan the entire screen, even in fit-to-screen mode.

**Comment by iiordanov** [comment]
It would have to be implemented quite differently because "Fit-To-Screen" really means no panning! :)

I'll think about how it could be implemented.

**Comment by freebrowser1** [comment]
> It would have to be implemented quite differently because "Fit-To-Screen" really means no panning! :)
> 
> I'll think about how it could be implemented.

Well, in the 'simulated touchpad' mode, swap two finger scroll with one-finger scroll.
That resolves another issue: every time when scrolling with two fingers, the two fingers click is also interpreted as a right mouse click so sometimes a context menu popup while touching the screen with two fingers when continuing scrolling.
When swapped, this side effect only occurs when panning the screen which occurs less often.

Another issue is scroll speed which is usually too fast even when scroll speed is set to the slowest in the target platform.

**Comment by iiordanov** [comment]
Maybe with two new input modes:

- Direct, Two-finger Pan
- Simulated Touchpad, Two-finger Pointer Move
